### PR TITLE
feat(taskframework): database change be executed  in TaskExecutor

### PR DESCRIPTION
**Pull request by krihy**
#### What type of PR is this?

type-feature
module-taskframework

#### What this PR does / why we need it:
1.  add `DatabaseChangeTask` for database change task be executed in TaskExecutor
2. task result of `DatabaseChangeTask` upload to oss
3. base task provide life cycle method `onInit/onStart/onStop/onFail` for subtask to callback
4. `ObjectStorageHandler`  is used to upload task result and log to oss
5. `JobUtils` provide getMetaDBConnectionConfig to DML storage some data to meta db

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/caller/NativeK8sJobClient.java [hunk 1]
```diff
@@ -51,15 +60,27 @@
...
+            }
+        } else {
+            apiClient = Config.defaultClient().setBasePath(k8sProperties.getUrl());
+        }
+        Verify.notNull(apiClient, "k8s api client");
         apiClient.setHttpClient(apiClient
                 .getHttpClient()
                 .newBuilder()
-                .readTimeout(Duration.ZERO)
+                .readTimeout(60000, TimeUnit.MILLISECONDS)
```
Comment (smallsheeeep) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/caller/NativeK8sJobClient.java:79]: Avoid using magic number, better to declare a static field.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/config/TaskFrameworkConfiguration.java [hunk 2]
```diff
@@ -46,8 +48,8 @@ public class TaskFrameworkConfiguration {
     public K8sJobClient k8sJobClient(@Autowired TaskFrameworkProperties taskFrameworkProperties) {
         if (taskFrameworkProperties.getRunMode() == TaskRunModeEnum.K8S) {
             try {
-                log.info("k8s url is {}", taskFrameworkProperties.getK8s().getUrl());
-                return new NativeK8sJobClient(taskFrameworkProperties.getK8s().getUrl());
+                log.info("k8s is not null {}", taskFrameworkProperties.getK8s());
```
Comment (smallsheeeep) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/config/TaskFrameworkConfiguration.java:51]: The log print and code is tricky

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/constants/JobDataMapConstants.java [hunk 3]
```diff
@@ -28,4 +31,16 @@ public class JobDataMapConstants {
     public static final String META_DB_TASK_PARAMETER = "metaDbTaskParameter";
 
     public static final String BUZ_ID = "buzId";
+
+    public static final String FLOW_INSTANCE_ID = "flowInstanceId";
+
+    public static final String CURRENT_SCHEMA_KEY = ConnectionSessionConstants.CURRENT_SCHEMA_KEY;
```
Comment (smallsheeeep) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/constants/JobDataMapConstants.java:37]: Why declare a constant related to another constant class?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/constants/JobDataMapConstants.java [hunk 4]
```diff
@@ -28,4 +31,16 @@ public class JobDataMapConstants {
...
+
+    public static final String SESSION_TIME_ZONE = ConnectionSessionConstants.SESSION_TIME_ZONE;
```
Comment (smallsheeeep) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/constants/JobDataMapConstants.java:39]: ditto

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/constants/JobDataMapConstants.java [hunk 5]
```diff
@@ -28,4 +31,16 @@ public class JobDataMapConstants {
...
+
+    public static final String OBJECT_STORAGE_CONFIGURATION = "objectStorageConfiguration";
+
+    public static final String OBJECT_METADATA = "objectMetadata";
+
+    public static final String TIMEOUT_MILLI_SECONDS = RuntimeTaskConstants.TIMEOUT_MILLI_SECONDS;
```
Comment (smallsheeeep) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/constants/JobDataMapConstants.java:45]: ditto

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/BaseTask.java [hunk 6]
```diff
@@ -39,29 +41,28 @@
...
-    protected volatile double progress = 0;
-
-    protected volatile boolean canceled = false;
-
-    private volatile boolean finished = false;
-
     private static final int REPORT_TASK_INFO_INTERVAL_SECONDS = 5;
     private static final int REPORT_RESULT_RETRY_TIMES = 10;
     private static final int REPORT_RESULT_RETRY_INTERVAL_SECONDS = 10;
+    private static final int DEFAULT_TASK_TIMEOUT_MILLI_SECONDS = 48 * 60 * 60 * 1000;
```
Comment (smallsheeeep) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/BaseTask.java:47]: MILLIS

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/BaseTask.java [hunk 7]
```diff
@@ -166,13 +160,15 @@ private void doFinal() {
     }
 
     private void reportTaskResult() {
-        onUpdate();
-        if (this.status == TaskStatus.DONE) {
-            this.progress = 1.0;
+        double progress = getProgress();
+        // onUpdate();
+        if (getTaskStatus() == TaskStatus.DONE) {
```
Comment (smallsheeeep) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/BaseTask.java:165]: The progress doesn't work within the buildCurrentResult() method

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/BaseTask.java [hunk 8]
```diff
@@ -212,4 +208,30 @@ private DefaultTaskResult buildCurrentResult() {
...
+
+    /**
+     * Deal with task run logic here
+     */
+    protected abstract void onStart();
+
+
+
+    /**
+     * Deal with task run logic here
```
Comment (smallsheeeep) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/BaseTask.java:226]: Bad comments

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/BaseTask.java [hunk 9]
```diff
@@ -212,4 +208,30 @@ private DefaultTaskResult buildCurrentResult() {
...
+     */
+    protected abstract void onStop();
+
+    /**
+     * Deal with task stop logic here
```
Comment (smallsheeeep) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/BaseTask.java:231]: ditto

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/DatabaseChangeTask.java [hunk 10]
```diff
@@ -0,0 +1,499 @@
...
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author wenniu.ly
+ * @date 2021/3/20
+ */
+
+@Slf4j
+public class DatabaseChangeTask extends BaseTask {
```
Comment (smallsheeeep) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/DatabaseChangeTask.java:90]: Good job👍

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/TaskFactory.java [hunk 11]
```diff
@@ -18,10 +18,13 @@
 
 import com.oceanbase.odc.service.task.executor.executor.TaskRuntimeException;
 
+import lombok.extern.slf4j.Slf4j;
+
 /**
  * @author gaoda.xy
  * @date 2023/11/24 11:01
  */
+@Slf4j
```
Comment (smallsheeeep) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/TaskFactory.java:27]: Seems no log print inner

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/util/JobUtils.java [hunk 12]
```diff
@@ -52,11 +53,25 @@ public static String toJson(Object obj) {
...
-        return port == null ? 8989 : Integer.parseInt(port);
+        return port != null ? Integer.parseInt(port) : 8989;
     }
 
     public static String getLogPath() {
         String logPath = SystemUtils.getEnvOrProperty(JobEnvConstants.LOG_DIRECTORY);
         return logPath == null ? "./log" : logPath;
     }
+
+    public static ConnectionConfig getMetaDBConnectionConfig() {
```
Comment (smallsheeeep) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/util/JobUtils.java:64]: Why executor need MetaDB config?

Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/util/JobUtils.java:64]: DLM need metadb to save data when user restart job

File: server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/DatabaseChangeRuntimeFlowableTaskCopied.java [hunk 13]
```diff
@@ -60,8 +67,7 @@ public class DatabaseChangeRuntimeFlowableTaskCopied extends BaseODCFlowTaskDele
     private volatile JobEntity jobEntity;
     private volatile boolean isSuccessful = false;
     private volatile boolean isFailure = false;
-    @Autowired
-    private CloudObjectStorageService cloudObjectStorageService;
+    private volatile DelegateExecution execution;
```
[the remark this answers is not in the document]
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/DatabaseChangeRuntimeFlowableTaskCopied.java:70]: ok ,done

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/BaseTask.java (hunk 8 above)
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/BaseTask.java]: removed it

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/TaskFactory.java (hunk 11 above)
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/TaskFactory.java]: done

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/caller/NativeK8sJobClient.java (hunk 1 above)
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/caller/NativeK8sJobClient.java]: done

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/BaseTask.java (hunk 9 above)
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/BaseTask.java]: done

File: server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/DatabaseChangeRuntimeFlowableTaskCopied.java [hunk 14]
```diff
@@ -118,7 +127,7 @@ protected DatabaseChangeResult start(Long taskId, TaskService taskService, Deleg
         return result;
     }
 
-    private JobDefinition buildJobDefinition(DelegateExecution execution) {
+    private JobDefinition buildJobDefinition() {
```
[the remark this answers is not in the document]
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/DatabaseChangeRuntimeFlowableTaskCopied.java:130]: done

File: server/odc-server/src/main/resources/data.sql [hunk 15]
```diff
@@ -711,5 +711,7 @@ INSERT INTO config_system_configuration ( `key`, `value`, `description` ) VALUES
 -- v4.2.4
 --
 INSERT INTO config_system_configuration ( `key`, `value`, `description` ) VALUES( 'odc.task-framework.run-mode', 'THREAD', 'ODC task run mode contain: THREAD/K8S, default is THREAD' ) ON DUPLICATE KEY UPDATE `id` = `id`;
-INSERT INTO config_system_configuration ( `key`, `value`, `description` ) VALUES( 'odc.task-framework.k8s.url', '', 'k8s url when task run mode is K8S' ) ON DUPLICATE KEY UPDATE `id` = `id`;
+INSERT INTO config_system_configuration ( `key`, `value`, `description` ) VALUES( 'odc.task-framework.odc-url', '', 'odc server url, constituted by domain or k8s service name, will use odc server ip when it is null') ON DUPLICATE KEY UPDATE `id`=`id`;
+INSERT INTO config_system_configuration ( `key`, `value`, `description` ) VALUES( 'odc.task-framework.k8s.kube-config', '', 'use k8s kube config connect to k8s default, instead of use k8s url when kube config is null' ) ON DUPLICATE KEY UPDATE `id` = `id`;
+INSERT INTO config_system_configuration ( `key`, `value`, `description` ) VALUES( 'odc.task-framework.k8s.url', '', 'use k8s url connect to k8s when kube config is null' ) ON DUPLICATE KEY UPDATE `id` = `id`;
```
[the remark this answers is not in the document]
Comment (krihy) [reply server/odc-server/src/main/resources/data.sql]: done

**Pull request merged**